backend/test/features/support/steps.js

```diff
@@ -1,158 +0,0 @@
-// features/support/steps.js
-import { Given, When, Then, AfterAll } from 'cucumber'
-import { expect } from 'chai'
-// import { client } from '../../../src/activitypub/apollo-client'
-import { GraphQLClient } from 'graphql-request'
-import Factory from '../../../src/seed/factories'
-import { host } from '../../../src/jest/helpers'
-const debug = require('debug')('ea:test:steps')
-
-const factory = Factory()
-const client = new GraphQLClient(host)
-
-function createUser (slug) {
-  debug(`creating user ${slug}`)
-  return factory.create('User', {
-    name: slug,
-    email: 'example@test.org',
-    password: '1234'
-  })
-  // await login({ email: 'example@test.org', password: '1234' })
-}
-
-Given('our own server runs at {string}', function (string) {
-  // just documenation
-})
-
-Given('we have the following users in our database:', function (dataTable) {
-  return Promise.all(dataTable.hashes().map((user) => {
-    return createUser(user.Slug)
-  }))
-})
-
-When('I send a GET request to {string}', async function (pathname) {
-  const response = await this.get(pathname)
-  this.lastContentType = response.lastContentType
-
-  this.lastResponses.push(response.lastResponse)
-  this.statusCode = response.statusCode
-})
-
-When('I send a POST request with the following activity to {string}:', async function (inboxUrl, activity) {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-06T23:20:41Z" title="Tuesday, May 7th 2019, 1:20:41 am +02:00">May 7, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,25 +1,37 @@
 // features/support/steps.js
-import { Given, When, Then, AfterAll } from 'cucumber'
+import { Given, When, Then, BeforeAll } from 'cucumber'
 import { expect } from 'chai'
-// import { client } from '../../../src/activitypub/apollo-client'
 import { GraphQLClient } from 'graphql-request'
 import Factory from '../../../src/seed/factories'
 import { host } from '../../../src/jest/helpers'
-const debug = require('debug')('ea:test:steps')
 
 const factory = Factory()
 const client = new GraphQLClient(host)
+// setDefaultTimeout(60 * 1000)
 
-function createUser (slug) {
-  debug(`creating user ${slug}`)
-  return factory.create('User', {
-    name: slug,
-    email: 'example@test.org',
-    password: '1234'
-  })
+async function createUser (slug) {
+  const result = await client.request(`
+    query {
+      User(slug: "${slug}") {
+        slug
+      }
+    }
+  `)
+  if (result.User.length === 0) {
+    return factory.create('User', {
+      name: slug,
+      email: 'example@test.org',
+      password: '1234',
+      actorId: `http://localhost:3000/api/users/${slug}`
+    })
+  }
   // await login({ email: 'example@test.org', password: '1234' })
 }
 
+BeforeAll('Clean Database for fresh test setup', async function () {
```


Great! I did not know cucumber has `BeforeAll` similar to `Given` and `When` I thought we have to use hooks in `features/support/`.

Reply:
> [<img alt="Mastercuber" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mastercuber) **Authored by [Mastercuber](https://github.com/Mastercuber)**
_<time datetime="2019-05-10T22:31:11Z" title="Saturday, May 11th 2019, 12:31:11 am +02:00">May 11, 2019</time>_

---


Yes, it's very handy.